crates/common/src/config.rs

```diff
@@ -49,7 +69,32 @@ pub struct RpcConfig {
     #[serde(default = "default_max_subscriptions_per_connection")]
     pub max_subscriptions_per_connection: u32,
 }
-
+impl FromEnv for RpcConfig{
+    fn from_env() -> anyhow::Result<Self> {
+        let mut toml_string = "".to_string();
+        toml_string.push_str(&format!("bind_host = {}\n", std::env::var("RPC_BIND_HOST")?));
+        toml_string.push_str(&format!("bind_port = {}\n", std::env::var("RPC_BIND_PORT")?));
+        if let Ok(val) = std::env::var("RPC_MAX_CONNECTIONS") {
+            toml_string.push_str(&format!("max_connections = {}\n", val));
+        }
+        if let Ok(val) = std::env::var("RPC_MAX_REQUEST_BODY_SIZE") {
+            toml_string.push_str(&format!("max_request_body_size = {}\n", val));
+        }
+        if let Ok(val) = std::env::var("RPC_MAX_RESPONSE_BODY_SIZE") {
+            toml_string.push_str(&format!("max_response_body_size = {}\n", val));
+        }
+        if let Ok(val) = std::env::var("RPC_BATCH_REQUESTS_LIMIT") {
+            toml_string.push_str(&format!("batch_requests_limit = {}\n", val));
+        }
+        if let Ok(val) = std::env::var("RPC_ENABLE_SUBSCRIPTIONS") {
+            toml_string.push_str(&format!("enable_subscriptions = {}\n", val));
```

Comment:
why go this way instead of just returning Self {} with std::env::var's

if you can't figure our how to handle optional fields of configs here's what you can do:

```rust
let optional_env_var = std::env::var("SOME_OPTIONAL_ENV_VAR").ok();
```

Reply:
I wanted to use serde default values and not duplicate const function names, but we discussed with Erce to move on as you said